examples/stm32/nucleo-f746zg-baremetal/main.c

```diff
@@ -4,79 +4,65 @@
 #include "hal.h"
 #include "mongoose.h"
 
-#define LED1 PIN('B', 0)      // On-board LED pin (green)
-#define LED2 PIN('B', 7)      // On-board LED pin (blue)
-#define LED3 PIN('B', 14)     // On-board LED pin (red)
-#define BTN1 PIN('C', 13)     // On-board user button
+#define LED PIN('B', 7)       // On-board LED pin
 #define BLINK_PERIOD_MS 1000  // LED blinking period in millis
 
-static uint64_t s_ticks, s_exti;  // Counters, increased by IRQ handlers
-
-uint64_t mg_millis(void) {  // Declare our own uptime function
-  return s_ticks;           // Return number of milliseconds since boot
-}
-
-void HardFault_Handler(void) {                // Escalated fault handler
-  gpio_output(LED3);                          // Setup red LED
-  for (;;) spin(2999999), gpio_toggle(LED3);  // Blink LED infinitely
-}
-
+static uint64_t s_ticks;      // Milliseconds since boot
 void SysTick_Handler(void) {  // SyStick IRQ handler, triggered every 1ms
   s_ticks++;
 }
 
-void EXTI15_10_IRQHandler(void) {  // External interrupt handler
-  s_exti++;
-  if (EXTI->PR & BIT(PINNO(BTN1))) EXTI->PR = BIT(PINNO(BTN1));
-  gpio_write(LED1, gpio_read(BTN1));  // No debounce. Turn LED if button pressed
+uint64_t mg_millis(void) {  // Let Mongoose use our uptime function
+  return s_ticks;           // Return number of milliseconds since boot
 }
 
-void SystemInit(void) {                  // Called automatically by startup code
-  clock_init();                          // Set clock to 180MHz
-  SysTick_Config(SYS_FREQUENCY / 1000);  // Increment s_ticks every ms
+void mg_random(void *buf, size_t len) {  // Use on-board RNG
+  for (size_t n = 0; n < len; n += sizeof(uint32_t)) {
+    uint32_t r = rng_read();
+    memcpy((char *) buf + n, &r, n + sizeof(r) > len ? len - n : sizeof(r));
+  }
 }
```

Comment:
This should perhaps be added to the driver, as all code using it will need it. Besides MG_ENABLE_CUSTOM_RANDOM we could resort to the driver's function when MG_ENABLE_MIP is defined; if the driver does not have one it falls back to Mongoose default.